Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -1882,3 +1936,351 @@ def parse_date_string(date_string, date_format='%Y-%m-%dT%H:%M:%S'):
+    :rtype: ``dict``
+    """
+    if not 0 <= score <= 3:
+        raise DemistoException('illegal DBot score, expected 0-3, got `{}`'.format(score))
+    indicator_type_lower = indicator_type.lower()
+    if indicator_type_lower not in INDICATOR_TYPE_TO_CONTEXT_KEY:
+        raise DemistoException('illegal indicator type, expected one of {}, got `{}`'.format(
+            INDICATOR_TYPE_TO_CONTEXT_KEY.keys(), indicator_type_lower
+        ))
+    # handle files
+    if INDICATOR_TYPE_TO_CONTEXT_KEY[indicator_type_lower] == 'file':
+        indicator_type_lower = 'file'
+    dbot_entry = {
+        outputPaths['dbotscore']: {
+            'Indicator': indicator,
+            'Type': indicator_type_lower,
+            'Vendor': vendor,
+            'Score': score
+        }
+    }
+    if score == 3 and build_malicious:
+        dbot_entry.update(build_malicious_dbot_entry(indicator, indicator_type, vendor, description))
+    return dbot_entry
+
+
+def build_malicious_dbot_entry(indicator, indicator_type, vendor, description=None):
+    """ Build Malicious dbot entry
+    Examples:
+        >>> build_malicious_dbot_entry('8.8.8.8', 'ip', 'Vendor', 'Google DNS')
+        {'IP(val.Address && val.Address == obj.Address)': {'Malicious': {'Vendor': 'Vendor', 'Description': 'Google DNS\
+'}, 'Address': '8.8.8.8'}}
+
+        >>> build_malicious_dbot_entry('md5hash', 'MD5', 'Vendor', 'Malicious File')
+        {'File(val.MD5 && val.MD5 == obj.MD5 || val.SHA1 && val.SHA1 == obj.SHA1 || val.SHA256 && val.SHA256 == obj.SHA\
+256 || val.SHA512 && val.SHA512 == obj.SHA512 || val.CRC32 && val.CRC32 == obj.CRC32 || val.CTPH && val.CTPH == obj.CTP\
+H || val.SSDeep && val.SSDeep == obj.SSDeep)': {'Malicious': {'Vendor': 'Vendor', 'Description': 'Malicious File'}\
+, 'MD5': 'md5hash'}}
+
+    :type indicator: ``str``
+    :param indicator: Value (e.g. 8.8.8.8)
+
+    :type indicator_type: ``str``
+    :param indicator_type: e.g. 'IP'
+
+    :type vendor: ``str``
+    :param vendor: Integration ID
+
+    :type description: ``str``
+    :param description: Why it's malicious
+
+    :return: A malicious DBot entry
+    :rtype: ``dict``
+    """
+    indicator_type_lower = indicator_type.lower()
+    if indicator_type_lower in INDICATOR_TYPE_TO_CONTEXT_KEY:
+        key = INDICATOR_TYPE_TO_CONTEXT_KEY[indicator_type_lower]
+        # `file` indicator works a little different
+        if key == 'file':
+            entry = {
+                indicator_type.upper(): indicator,
+                'Malicious': {
+                    'Vendor': vendor,
+                    'Description': description
+                }
+            }
+            return {outputPaths[key]: entry}
+        else:
+            entry = {
+                key: indicator,
+                'Malicious': {
+                    'Vendor': vendor,
+                    'Description': description
+                }
+            }
+            return {outputPaths[indicator_type_lower]: entry}
```

Comment:
I think that you have indicators that are not in the outputPaths, make sure that they are all there.

Reply:
like what? all of the hashes I added, they are all in.